sympy/solvers/solveset.py

```diff
@@ -987,14 +987,25 @@ def _solveset(f, symbol, domain, _check=False):
 
 def term_factors(f):
     """
-    Returns a list of the factors of all terms present in the
-    expression.
+    An iterator that returns the factors of all terms
+    present in the expression.
+
+    Examples
+    ========
+
+    >>> from sympy import symbols
+    >>> from sympy.solvers.solveset import term_factors
+    >>> x = symbols('x')
+    >>> term_factors_obj = term_factors(x**2 - 1)
```

Comment:
```
>>> list(term_factors(-2 - x**2 + x*(x + 1)))
[-2, -1, x**2, x, x + 1]
```

Reply:
Should I remove the loop? It will probably demonstrate another way to use the results from the function rather than converting into a list or tuple